hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/ReadFilesP.java

```diff
@@ -57,9 +56,9 @@
 
     @Override
     public boolean complete() {
-        try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(directory, glob == null ? "*" : glob)) {
+        try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(directory, glob)) {
             StreamSupport.stream(directoryStream.spliterator(), false)
-                    .filter(this::shouldProcessEvent)
+                    .filter(this::shouldProcessEventAndNotDir)
```

Comment:
I don't understand this name. After studying its implementation for a while, looks like `filesAssignedToMe` or something similar.